src/dao/ContactDao.ts

```diff
@@ -0,0 +1,58 @@
+import Sanitizer from "../utils/Sanitizer";
+import Contact, { ContactType } from "../db/models/Contact";
+import QueryUtils from "../utils/queryUtils";
+import { ContactQueryOptions, ContactInput } from "../graphql/types/contactTypes";
+
+const ContactDao = {
+
+  async create(payload: Contact, userId: number) {
+    const {
+      type,
+      value,
+    } = payload;
+    if (!ContactType[type] || !Sanitizer.isValidStr(value)) {
+      return null;
+    }
+    if (userId == null) {
+      throw new Error("Creation failed: missing 'userId'");
+    }
+    return Contact.create({ ...payload, userId });
+  },
+
+  async update(payload: ContactInput, userId: number) {
+    if (!Sanitizer.isValidInt(payload?.id)) {
+      throw new Error("Invalid request content");
+    }
+    await Contact.update(payload, {
```

Comment:
The value returned is equal to the amount of rows updated - check its value, if it is equal to 0, then the update failed and you can throw an error

Reply:
its important to do it, because if we try to execute the query passing the token of someone else, then the userId might not match any record - then we will update 0 rows, so we should tell that in the response